File: skyspace999_ui/src/services/ConnectionFlightService.js
import http from "../http-common"; 

class ConnectionFlightService {
  getAllConnectionFlights(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/connectionFlight/connectionFlights`, searchDTO);
  }

  get(connectionFlightId) {
    return this.getRequest(`/connectionFlight/${connectionFlightId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/connectionFlight?field=${matchData}`, null);
  }

  addConnectionFlight(data) {
    return http.post("/connectionFlight/addConnectionFlight", data);
  }

  update(data) {
  	return http.post("/connectionFlight/updateConnectionFlight", data);
  }
  
  uploadImage(data,connectionFlightId) {
  	return http.postForm("/connectionFlight/uploadImage/"+connectionFlightId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ConnectionFlightService();

File: src/main/java/com/skyspace999/service/ConnectionFlightService.java
package com.skyspace999.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspace999.domain.ConnectionFlight;
import com.skyspace999.dto.ConnectionFlightDTO;
import com.skyspace999.dto.ConnectionFlightSearchDTO;
import com.skyspace999.dto.ConnectionFlightPageDTO;
import com.skyspace999.dto.ConnectionFlightConvertCriteriaDTO;
import com.skyspace999.service.GenericService;
import com.skyspace999.dto.common.RequestDTO;
import com.skyspace999.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ConnectionFlightService extends GenericService<ConnectionFlight, Integer> {

	List<ConnectionFlight> findAll();

	ResultDTO addConnectionFlight(ConnectionFlightDTO connectionFlightDTO, RequestDTO requestDTO);

	ResultDTO updateConnectionFlight(ConnectionFlightDTO connectionFlightDTO, RequestDTO requestDTO);

    Page<ConnectionFlight> getAllConnectionFlights(Pageable pageable);

    Page<ConnectionFlight> getAllConnectionFlights(Specification<ConnectionFlight> spec, Pageable pageable);

	ResponseEntity<ConnectionFlightPageDTO> getConnectionFlights(ConnectionFlightSearchDTO connectionFlightSearchDTO);
	
	List<ConnectionFlightDTO> convertConnectionFlightsToConnectionFlightDTOs(List<ConnectionFlight> connectionFlights, ConnectionFlightConvertCriteriaDTO convertCriteria);

	ConnectionFlightDTO getConnectionFlightDTOById(Integer connectionFlightId);







}





